src/main/java/com/hivemq/cli/commands/hivemq/export/clients/ClientDetailsCsvWriterTask.java

```diff
@@ -0,0 +1,288 @@
+/*
+ * Copyright 2019 HiveMQ and the HiveMQ Community
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package com.hivemq.cli.commands.hivemq.export.clients;
+
+import com.hivemq.cli.openapi.hivemq.CertificateInformation;
+import com.hivemq.cli.openapi.hivemq.ClientDetails;
+import com.hivemq.cli.openapi.hivemq.ClientRestrictions;
+import com.hivemq.cli.openapi.hivemq.ConnectionDetails;
+import com.hivemq.cli.openapi.hivemq.ProxyInformation;
+import com.hivemq.cli.openapi.hivemq.TLV;
+import com.hivemq.cli.openapi.hivemq.TlsInformation;
+import com.opencsv.CSVWriter;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.Callable;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+
+public class ClientDetailsCsvWriterTask implements Callable<Void> {
+
+    public static final String[] EXPORT_CSV_HEADER = {
+            "clientId",
+            "connected",
+            "sessionExpiryInterval",
+            "connectedAt",
+            "messageQueueSize",
+            "willPresent",
+            "maxMessageSize",
+            "maxQueueSize",
+            "queuedMessageStrategy",
+            "ip",
+            "sourceIp",
+            "sourcePort",
+            "destinationIp",
+            "destinationPort",
+            "tlvs",
+            "mqttVersion",
+            "connectedListenerId",
+            "connectedNodeId",
+            "keepAlive",
+            "username",
+            "password",
+            "cleanStart",
+            "cipherSuite",
+            "tlsVersion",
+            "certificateCommonName",
+            "certificateOrganization",
+            "certificateOrganizationalUnit",
+            "certificateSerial",
+            "certificateValidFrom",
+            "certificateValidUntil",
+            "certificateCountry",
+            "certificateState"
+    };
+
+    final @NotNull Future<Void> clientDetailsFuture;
+    final @NotNull BlockingQueue<ClientDetails> clientDetailsQueue;
+    final @NotNull File file;
+    final @NotNull CSVWriter csvWriter;
+    final @NotNull BufferedWriter bufferedFileWriter;
+
+    long writtenClientDetails = 0;
```

Comment:
Needs to be at least volatile depending on the consistency guarantees you want to give.